test/platform-checks/mzcompose.py

```diff
@@ -73,7 +75,9 @@ def setup(c: Composition) -> None:
     c.up("testdrive", persistent=True)
     c.up("cockroach")
 
-    c.up("redpanda", "postgres", "debezium", "minio")
+    c.up(
+        "redpanda", "postgres", "debezium", "minio", "ssh-bastion-host", "materialized"
+    )
```

Comment:
something about this change (adding `materialized`) i think is causing https://github.com/MaterializeInc/materialize/pull/22969

perhaps because `StartMz` is overriding it or something?

Reply:
perhaps the external cockroach is not quite up yet, and we are spitting out a weird error for it: 
![image](https://github.com/MaterializeInc/materialize/assets/5404303/4fe4c866-b8d5-4a0f-a6de-8ab8938ec1dd)

cc @ParkMyCar 

We seem to be running migrations in a retry-loop, and it cockroach isnt available (or is temporarily available and then becomes unavailable again during startup or something), the retry loop sees a weird state

Reply:
also `Cockroach` is starting up in memory-only mode, so it could definitely be doing something weird or losing data

i think the solution _might_ be to wait for `Cockroach` to actually startup after doing `up`

Reply:
I'm not sure if this is an issue with retries, because if we can read a Stash version then it means we've connected to Cockroach and read from our collections.

Something else I just noticed which is odd, the minimum version is reported as 25. The [PR](https://github.com/MaterializeInc/materialize/pull/22201) that bumped this to the current minimum (35) went out with v0.73. So it would seem to me that we're initializing the Stash with a MZ v0.72 (or maybe older), and then somehow we come back online with the current version.

Reply:
Oh no, probably adding `materialized` here is wrong indeed. I'll try to remove it.